elastic/kibana
# [Index Management] Support data retention on Data Streams tab

This PR adds support for viewing the configured data retention value per data stream on the Data Streams tab in Index Management.

On **serverless**, a data stream can be managed by data stream lifecycle.

On **stateful**, a data stream can be managed by data stream lifecycle or ILM, or both. 

Fixes https://github.com/elastic/kibana/issues/165134
Fixes https://github.com/elastic/kibana/issues/152720
Fixes https://github.com/elastic/kibana/issues/154259
Fixes https://github.com/elastic/kibana/issues/165143

Note: Skipping release note for now, however, we will want to add one once the remaining work is complete (meta: https://github.com/elastic/kibana/issues/154256)

### How to test
1. Create a data stream managed via data stream lifecycle ([docs](https://www.elastic.co/guide/en/elasticsearch/reference/master/tutorial-manage-new-data-stream.html)). The configured data retention should appear on the list view and details flyout.
2. [Stateful only] Create a data stream managed via ILM ([docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/set-up-a-data-stream.html)). There should be no data retention value. On the details flyout, there should be a link to the ILM policy.
3. [Stateful only] Create a data stream managed by ILM and data stream lifecycle. The configured data retention should appear on the list view and details flyout. The ILM policy name should also appear on the details flyout.

### Screenshots
<img width="1212" alt="Screenshot 2023-09-06 at 3 49 54 PM" src="https://github.com/elastic/kibana/assets/5226211/b51ef882-6787-4902-8b03-a7a2cf93f208">

<img width="274" alt="Screenshot 2023-09-06 at 12 52 03 PM" src="https://github.com/elastic/kibana/assets/5226211/e140dd11-7528-4051-a6eb-70e58f7fcd25">

<img width="509" alt="Screenshot 2023-09-06 at 12 52 10 PM" src="https://github.com/elastic/kibana/assets/5226211/a9cccd3c-5556-4245-8591-cfbf0d2a6404">

<img width="268" alt="Screenshot 2023-09-06 at 12 52 15 PM" src="https://github.com/elastic/kibana/assets/5226211/20dcbd39-3ab1-434c-8e60-f6ea476d3ea6">
